src/client/Microsoft.Identity.Client/PlatformsCommon/Shared/InMemoryPartitionedAppTokenCacheAccessor.cs

```diff
@@ -175,6 +175,7 @@ public void DeleteAccount(MsalAccountCacheItem item)
         /// </summary>
         public virtual IReadOnlyList<MsalAccessTokenCacheItem> GetAllAccessTokens(string partitionKey = null)
         {
+            _logger.LogAlways($"Total number of cache partitions found while getting access tokens: {AccessTokenCacheDictionary.Count}");
```

Comment:
This might not be accurate in all of scenarios, like shared cache, because when the last cache item is removed from the partition dictionary, the dictionary itself is not removed. It's accurate for serialization scenario since we clear the dictionaries. We can probably update the removal logic, to remove empty partition. @bgavrilMS

Reply:
I think it makes sense to remove the empty partitions

Reply:
Based on the discussion, this is fine then; in a shared cache we would remove token then re-add with the same key.